File: AugustineKadima/StacksAndQueues/MyQueue.java
public class MyQueue {
    private static int[] randomQueue = new int[0];
    private static int front, back,size = 0;

    public MyQueue(int Size){
        randomQueue = new int[Size];
    }

    public void push(int element){
        if(randomQueue.length > size){
            randomQueue[back] = element;
            ++back;
            ++size;
        }else{
            System.out.println("The queue is full! Wait for a dequeue.");
        }
    }

    public int pop(){
        int frontValue = 0;
        if(front == back){
            frontValue = -1;
        }else{
            frontValue = randomQueue[front];

        }


        for(int i = 0; i < size; i++){
            if(i < size -1 && back >= 1) {
                randomQueue[i] = randomQueue[i + 1];
            }
        }
        randomQueue[size - 1] = 0;
        --size;
        --back;

        return frontValue;
    }

    public int peek(){
        return randomQueue[front];
    }

    public boolean empty(){
        return size == 0;
    }

    public static void main(String[] args) {
        MyQueue athletes = new MyQueue(5);
        athletes.push(1);
        athletes.push(2);
        athletes.push(5);
        athletes.push(4);
        athletes.push(3);

        athletes.pop();
        athletes.pop();
        athletes.pop();


        athletes.push(7);



        for(int element:MyQueue.randomQueue){
            System.out.println(element);
        }

        System.out.println("The peek is: "+athletes.peek());
        System.out.println("Is array empty: "+ athletes.empty());

    }


}

File: Shizler/Stacks and Queues/Queues/myQueue.py
# Implement a first in first out (FIFO) queue using only two stacks. The implemented queue should support all the functions of a normal queue (push, peek, pop, and empty).

# Implement the MyQueue class:
	
# void push(int x) Pushes element x to the back of the queue.
	
# int pop() Removes the element from the front of the queue and returns it.
	
# int peek() Returns the element at the front of the queue.
	
# boolean empty() Returns true if the queue is empty, false otherwise.
class MyQueue:
    def __init__(self) -> None:
        self.stackInput = []
        self.stackOutput = []
    def push(self,val):
        self.stackInput.append(val)
    def pop(self):
        if not self.stackOutput:
            while self.stackInput:
                self.stackOutput.append(self.stackInput.pop())
        return self.stackOutput.pop()
    def peek(self):
        if not self.stackOutput:
            while self.stackInput:
                self.stackOutput.append(self.stackInput.pop())
        return self.stackOutput[-1]
    def empty(self):
        return not self.stackInput and not self.stackOutput


if __name__ == '__main__':
    que = MyQueue()
    que.push(42)
    print(que.pop()) #42
    que.push(14)
    print(que.empty()) #False
    print(que.peek())  #14
    que.push(28)
    que.push(60)
    print(que.empty()) #False
    print(que.pop()) #14